contracts/cw-proposal-multiple/src/contract.rs

```diff
@@ -261,7 +283,13 @@ pub fn execute_vote(
     )?;
 
     let old_status = prop.status;
-    prop.votes.add_vote(vote, vote_power);
+
+    // Remove the old vote if this is a re-vote.
+    if let Some(ballot) = previous_ballot {
+        prop.votes.remove_vote(ballot.vote, ballot.power)?;
```

Comment:
can we just do `remove_vote` in the `BALLOTS.update` closure above since we don't seem to need `previous_ballet` otherwise?

Reply:
Indeed, I've updated it. Also changed the prop-single for consistency.